packages/react/src/components/SpinButton/SpinButton.styles.ts

```diff
@@ -192,49 +191,34 @@ export const getStyles = (props: ISpinButtonStyleProps): ISpinButtonStyles => {
         boxSizing: 'border-box',
         height: DEFAULT_HEIGHT,
         minWidth: DEFAULT_MIN_WIDTH,
-        selectors: {
-          // setting border using pseudo-element here in order to prevent:
-          // input and chevron buttons to overlap border under certain resolutions
-          ':after': {
-            pointerEvents: 'none',
-            content: "''",
-            position: 'absolute',
-            left: 0,
-            top: 0,
-            bottom: 0,
-            right: 0,
-            borderWidth: '1px',
-            borderStyle: 'solid',
-            borderColor: SpinButtonRootBorderColor,
-            borderRadius: effects.roundedCorner2,
+        ...getInputFocusStyle(SpinButtonRootBorderColor, effects.roundedCorner2),
+        ':after': {
+          borderWidth: '1px',
+          [HighContrastSelector]: {
+            borderColor: 'GrayText',
           },
         },
       },
       (labelPosition === Position.top || labelPosition === Position.bottom) && {
         width: '100%',
       },
+
       !disabled && [
         {
-          selectors: {
-            ':hover': {
-              selectors: {
-                ':after': {
-                  borderColor: SpinButtonRootBorderColorHovered,
-                },
-                [HighContrastSelector]: {
-                  selectors: {
-                    ':after': {
-                      borderColor: 'Highlight',
-                    },
-                  },
-                },
-              },
+          ':hover:after': {
+            borderColor: SpinButtonRootBorderColorHovered,
+            [HighContrastSelector]: {
+              borderColor: 'Highlight',
             },
           },
         },
         isFocused && {
-          selectors: {
-            '&&': getInputFocusStyle(SpinButtonRootBorderColorFocused, effects.roundedCorner2),
+          ':hover:after, :after': {
```

Comment:
move away from duplicative border styles, and only specifying what has changed. remove the && which duplicates the className, and override the :hover:after, as well as the after in this selector

Reply:
What happens if someone bet on the duplicated class name for their existing custom styling?

Reply:
i agree, that any css change could break someone with some edge usecase. but the issue here is that to modify the focused styles, the user has to know how to write a duplicated className (the &&).